### ISSUE TITLE
pdf module bug: when i removed some pages from pdf, then save it. the new file's size is almost the same as the old file's.

### ISSUE THREAD

**Issue by taihangg** [syncfusion/flutter-widgets#2049]
when i removed some pages from pdf, then save it. the new file's size is almost the same as the old file's.

**Comment by jeyalakshmit** [comment]
<html>
<body>
<span data-teams="true"><span class="ui-provider a b c d e f g h i j k l m n o p q r s t u v w x y z ab ac ae af ag ah ai aj ak" dir="ltr"><p>By default, when updating a PDF document, we use incremental updates. This means that when pages are removed from an existing PDF file, the original content remains unchanged, and the updated content is added after it. As a result, the file size does not decrease when pages are removed. To resolve this, you can disable the <a aria-label="Link incrementalUpdate" id="menurgtl" href="https://pub.dev/documentation/syncfusion_flutter_pdf/latest/pdf/PdfFileStructure/incrementalUpdate.html" rel="noreferrer noopener" target="_blank" class="fui-Link ___1q1shib f2hkw1w f3rmtva f1ewtqcl fyind8e f1k6fduh f1w7gpdv fk6fouc fjoy568 figsok6 f1s184ao f1mk8lai fnbmjn9 f1o700av f13mvf36 f1cmlufx f9n3di6 f1ids18y f1tx3yz7 f1deo86v f1eh06m1 f1iescvh fhgqx19 f1olyrje f1p93eir f1nev41a f1h8hb77 f1lqvz6u f10aw75t fsle3fq f17ae5zn" title="https://pub.dev/documentation/syncfusion_flutter_pdf/latest/pdf/pdffilestructure/incrementalupdate.html"><span>incrementalUpdate</span></a> property available in the <a aria-label="Link FileStructure" id="menurgtn" href="https://pub.dev/documentation/syncfusion_flutter_pdf/latest/pdf/PdfFileStructure-class.html" rel="noreferrer noopener" target="_blank" class="fui-Link ___1q1shib f2hkw1w f3rmtva f1ewtqcl fyind8e f1k6fduh f1w7gpdv fk6fouc fjoy568 figsok6 f1s184ao f1mk8lai fnbmjn9 f1o700av f13mvf36 f1cmlufx f9n3di6 f1ids18y f1tx3yz7 f1deo86v f1eh06m1 f1iescvh fhgqx19 f1olyrje f1p93eir f1nev41a f1h8hb77 f1lqvz6u f10aw75t fsle3fq f17ae5zn" title="https://pub.dev/documentation/syncfusion_flutter_pdf/latest/pdf/pdffilestructure-class.html"><span>FileStructure</span></a> API.</p><p>Additionally, we recommend setting the cross-reference type to a stream instead of a table to further reduce the file size. Please refer to the following code example for more details.</p>

```
//Disable the incremental update
document.fileStructure.incrementalUpdate = false;
//Set the cross reference type as cross reference stream
document.fileStructure.crossReferenceType = PdfCrossReferenceType.crossReferenceStream;
```


<p>&nbsp;</p><p>For more information, please refer to the user guide and API documentation:</p> https://help.syncfusion.com/document-processing/pdf/pdf-library/flutter/working-with-document#performing-incremental-update-for-the-pdf-document</p>
<p>Please try this solution, and if you continue to experience issues, we kindly request that you share the input PDF document with us. This will allow us to analyze the situation and provide further assistance.</p></span></span>
</body>
</html>

**Comment by taihangg** [comment]
Thank you!.I have tried the example code, but it does not as expected.
my code as this:

```
int startPage=3, endPage=5;

final _pdfDoc = PdfDocument(inputBytes: File(xxx).readAsBytesSync());
_pdfDoc!.fileStructure.incrementalUpdate = false;
_pdfDoc!.fileStructure.crossReferenceType =
    PdfCrossReferenceType.crossReferenceStream;

for (int i = _pageCount! - 1; endPage! <= i; i--) {
  _pdfDoc!.pages.removeAt(i);
}
for (int i = startPage - 2; 0 <= i; i--) {
  _pdfDoc!.pages.removeAt(i);
}

await saveFile(newPath, _pdfDoc!.save());
_pdfDoc!.dispose();
```
An，this is my file：
[成功家教启示录 -- 教育部关工委家教中心编写 -- 2001 -- 南京_南京大学出版社.pdf](https://github.com/user-attachments/files/16882444/--.--.2001.--._.pdf)

**Comment by irfanajaffer** [comment]
<html>
<body>
<span data-teams="true"><span class="ui-provider a b c d e f g h i j k l m n o p q r s t u v w x y z ab ac ae af ag ah ai aj ak" dir="ltr"><p style="margin-bottom: 8px;">We have reproduced the reported behavior and are currently analyzing it. Further details will be provided on September 10, 2024.</p><p style="margin-bottom: 8px;">In the meantime, you can use our <a aria-label="Link createTemplate" id="menur9iv" href="https://pub.dev/documentation/syncfusion_flutter_pdf/latest/pdf/PdfPage/createTemplate.html" rel="noreferrer noopener" target="_blank" class="fui-Link ___1q1shib f2hkw1w f3rmtva f1ewtqcl fyind8e f1k6fduh f1w7gpdv fk6fouc fjoy568 figsok6 f1s184ao f1mk8lai fnbmjn9 f1o700av f13mvf36 f1cmlufx f9n3di6 f1ids18y f1tx3yz7 f1deo86v f1eh06m1 f1iescvh fhgqx19 f1olyrje f1p93eir f1nev41a f1h8hb77 f1lqvz6u f10aw75t fsle3fq f17ae5zn" title="https://pub.dev/documentation/syncfusion_flutter_pdf/latest/pdf/pdfpage/createtemplate.html">createTemplate</a> method on the loaded page to achieve the desired optimized file size. Please refer to the following code snippet for guidance:</p>


<p style="margin-bottom: 8px;">&nbsp;</p><p style="margin-bottom: 8px;">For more information on working with pages, kindly use the following link: <a aria-label="Link Pages in Flutter PDF library | Syncfusion" id="menur9j1" href="https://help.syncfusion.com/document-processing/pdf/pdf-library/flutter/working-with-pages" rel="noreferrer noopener" target="_blank" class="fui-Link ___1q1shib f2hkw1w f3rmtva f1ewtqcl fyind8e f1k6fduh f1w7gpdv fk6fouc fjoy568 figsok6 f1s184ao f1mk8lai fnbmjn9 f1o700av f13mvf36 f1cmlufx f9n3di6 f1ids18y f1tx3yz7 f1deo86v f1eh06m1 f1iescvh fhgqx19 f1olyrje f1p93eir f1nev41a f1h8hb77 f1lqvz6u f10aw75t fsle3fq f17ae5zn" title="https://help.syncfusion.com/document-processing/pdf/pdf-library/flutter/working-with-pages">Pages in Flutter PDF library | Syncfusion</a></p></span></span>
</body>
</html>

```
 int startPage = 3, endPage = 5;
 //Load the existing the document
 PdfDocument existingDocument =
     PdfDocument(inputBytes: File(xxx).readAsBytesSync());
 //Create a new PDF document
 PdfDocument newDocument = PdfDocument();
 //Initialize compression level
 newDocument.compressionLevel = PdfCompressionLevel.normal;
 //Set the cross-reference type as cross reference stream
  newDocument.fileStructure.crossReferenceType =
      PdfCrossReferenceType.crossReferenceStream;
 //Set page margins to zero.
 newDocument.pageSettings.margins.all = 0;
 //Export the required pages to the new document.
 for (int pageIndex = startPage - 1; pageIndex < endPage; pageIndex++) {
   //Load the existing page as template.
   PdfTemplate pageTemplate =
        existingDocument.pages[pageIndex].createTemplate();
   //Create a page in the new document.
   PdfPage page = newDocument.pages.add();
   //Draw the existing page template in the new document.
   page.graphics
       .drawPdfTemplate(pageTemplate, Offset.zero, page.getClientSize());
 }
 //Save
 await saveFile(newPath, newDocument!.save());
 //Dispose the documents
 newDocument.dispose();
 existingDocument.dispose();
 ```

**Comment by taihangg** [comment]
Thank you very much!
This works!

**Comment by taihangg** [comment]
excuse me, can i close this issue now?
or wait for a time.

**Comment by taihangg** [comment]
excuse me, can i close this issue now?
or wait for a time.

**Comment by jeyalakshmit** [comment]
We have confirmed the issue ”File size does not decrease after removing pages from the PDF document” as a defect in our product and fix will be included in our upcoming weekly release, which will be available on September 24, 2024.

Use the below feedback link to track the status of the reported bug.
[File size does not decrease after removing pages from the PDF document in Flutter | Feedback Portal (syncfusion.com)](https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.syncfusion.com%2Ffeedback%2F61120%2Ffile-size-does-not-decrease-after-removing-pages-from-the-pdf-document&data=05%7C02%7Cjeyalakshmi.t%40syncfusion.com%7C8e1791224e934822827808dcd19c22ba%7C77f1fe12b04949198c509fb41e5bb63b%7C0%7C0%7C638615715729452074%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=qrFE0Hy1w3Vy5S7Hizraweh9MmYziNzbhMqbqMJXzdQ%3D&reserved=0)

Disclaimer: “Inclusion of this solution in the weekly release may change due to other factors including but not limited to QA checks and works reprioritization.”

**Comment by taihangg** [comment]
that's very good!thank you!

**Comment by irfanajaffer** [comment]
The fix for the reported issue "File size does not decrease after removing pages from the PDF document" has been included in our weekly NuGet release v27.1.50.

 

Kindly use the following link to refer to the latest release version,

[syncfusion_flutter_pdf | Flutter package (pub.dev)](https://nam02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fpub.dev%2Fpackages%2Fsyncfusion_flutter_pdf%2Fversions%2F27.1.50&data=05%7C02%7Cirfana.jaffersadhik%40syncfusion.com%7C9b65dda981f34e4b137f08dcdc936d43%7C77f1fe12b04949198c509fb41e5bb63b%7C0%7C0%7C638627772969779867%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=ZnuBy3mL%2FALTvjJ5kXwDwg5SO3BBjm45aiuzdZMrSGk%3D&reserved=0)
